packages/nx/src/generators/utils/glob.ts

```diff
@@ -0,0 +1,63 @@
+import { Tree } from '../tree';
+import { combineGlobPatterns } from '../../utils/globs';
+import { workspaceRoot } from '../../utils/workspace-root';
+
+import minimatch = require('minimatch');
+
+/**
+ * Performs a glob search on the files in a workspace. Paths should be unix-style
+ * with forward slashes.
+ *
+ * @param patterns A list of glob patterns
+ * @returns Normalized paths in the workspace that match the provided glob patterns.
+ */
+export async function glob(patterns: string[]): Promise<string[]>;
+
+/**
+ * Performs a tree-aware glob search on the files in a workspace. Able to find newly
+ * created files and hides deleted files before the updates are committed to disk.
+ * Paths should be unix-style with forward slashes.
+ *
+ * @param tree The file system tree
+ * @param patterns A list of glob patterns
+ * @returns Normalized paths in the workspace that match the provided glob patterns.
+ */
+export async function glob(tree: Tree, patterns: string[]): Promise<string[]>;
+
+export async function glob(
+  treeOrPatterns: Tree | string[],
+  maybePatterns?: string[]
+): Promise<string[]> {
+  const { globWorkspace } = await import('../../native');
```

Comment:
This is the only reason why it's `async`? Can we make this a import at the top of the file?

Reply:
Ah, yeah we can move it to top. For some reason I thought we always used dynamic import for native.